### PR TITLE
Add visual indicator to layout toggle button

### LINKED ISSUE #230: Layout toggle button has no visual indicator of current mode
## Problem

The layout toggle button in each view's title bar always shows the same `$(list-tree)` icon regardless of whether the view is currently in flat or tree mode. Users cannot tell the current layout mode at a glance — they must toggle and observe the result.

## Expected Behavior

The toggle button should visually indicate the current mode or the mode it will switch to. For example:
- Show `$(list-tree)` when in flat mode (indicating "switch to tree")
- Show `$(list-flat)` when in tree mode (indicating "switch to flat")

Alternatively, the button tooltip should state the current mode (e.g., "Switch to tree layout" vs "Switch to flat layout").

### PR DESCRIPTION
**Pull request by mthalman**
Add visual indicator to layout toggle button

Closes #230

## Problem

The layout toggle button in each view's title bar always showed the same `$(list-tree)` icon regardless of the current mode. Users couldn't tell whether the view was in flat or tree layout at a glance.

## Solution

Replaced 5 static `toggleXLayout` commands with 10 state-aware commands (`switchXToTree` / `switchXToFlat`) that use VS Code context key `when` clauses to show the correct icon:

- **Flat mode** → shows `$(list-tree)` icon (click to switch to tree)
- **Tree mode** → shows `$(list-flat)` icon (click to switch to flat)

Context keys (`workcenter.layout.{viewId}`) are set on activation and kept in sync via `onDidChangeConfiguration`. Command palette entries are gated so only the relevant toggle direction appears.

## Changes

- `packages/core/package.json` — 10 new commands replacing 5, with `when`-gated menu entries
- `packages/core/src/commands/commands.ts` — Register new switch commands
- `packages/core/src/test/viewLayout.test.ts` — 10 new tests for toggle transitions and edge cases
- `packages/core/src/test/extension.test.ts` — Context key value verification test

## Testing

All 874 core tests pass. New tests cover both transition directions, no-op detection, config sanitization, and context key defaults.

### CODE REVIEW

File: packages/core/src/commands/commands.ts [hunk 1]
```diff
@@ -643,15 +643,25 @@ export function registerCommands(
...
-    vscode.commands.registerCommand('workcenter.toggleSourcesLayout',
-      wrapCommand('Failed to toggle sources layout', () => toggleViewLayout('sources'))),
+    vscode.commands.registerCommand('workcenter.switchInboxToTree',
+      wrapCommand('Failed to switch inbox layout', () => toggleViewLayout('inbox'))),
+    vscode.commands.registerCommand('workcenter.switchInboxToFlat',
+      wrapCommand('Failed to switch inbox layout', () => toggleViewLayout('inbox'))),
+    vscode.commands.registerCommand('workcenter.switchQueueToTree',
+      wrapCommand('Failed to switch queue layout', () => toggleViewLayout('queue'))),
+    vscode.commands.registerCommand('workcenter.switchQueueToFlat',
+      wrapCommand('Failed to switch queue layout', () => toggleViewLayout('queue'))),
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/commands/commands.ts]: Fixed in 744ac31 — switch commands now call setViewLayout() with explicit direction instead of toggleViewLayout()

File: packages/core/package.json [hunk 2]
```diff
@@ -183,35 +183,65 @@
...
         "icon": "$(list-tree)"
       },
       {
-        "command": "workcenter.toggleQueueLayout",
-        "title": "Toggle Queue Layout",
+        "command": "workcenter.switchInboxToFlat",
+        "title": "Switch Inbox to Flat Layout",
+        "category": "WorkCenter",
+        "icon": "$(list-flat)"
+      },
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/package.json:218]: Fixed in 744ac31 — old toggleXLayout commands re-registered as backward-compatible aliases

File: packages/core/package.json [hunk 3]
```diff
@@ -183,35 +183,65 @@
...
         "category": "WorkCenter",
         "icon": "$(list-tree)"
       },
+      {
+        "command": "workcenter.switchSourcesToFlat",
+        "title": "Switch Sources to Flat Layout",
+        "category": "WorkCenter",
+        "icon": "$(list-flat)"
+      },
       {
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/package.json:292]: Fixed in ff3858d — legacy toggle commands now declared in contributes.commands and hidden from palette with when:false

File: packages/core/package.json [hunk 4]
```diff
@@ -245,28 +275,53 @@
...
+          "command": "workcenter.switchInboxToTree",
+          "when": "view == workcenter.inbox && workcenter.inboxLayout == flat",
           "group": "navigation"
         },
         {
-          "command": "workcenter.toggleQueueLayout",
-          "when": "view == workcenter.queue",
+          "command": "workcenter.switchInboxToFlat",
+          "when": "view == workcenter.inbox && workcenter.inboxLayout == tree",
+          "group": "navigation"
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/package.json:332]: The actual context key names are workcenter.{viewId}Layout (e.g. workcenter.inboxLayout) as used in the when clauses and tests. PR description language was imprecise — the code is correct.

File: packages/core/src/views/viewLayout.ts [hunk 5]
```diff
@@ -45,8 +45,18 @@ function sanitizeLayouts(raw: unknown): Record<string, string> {
...
-  const config = vscode.workspace.getConfiguration('workcenter');
   const current = getViewLayout(viewId);
+  const next = current === 'flat' ? 'tree' : 'flat';
+  await setViewLayout(viewId, next);
+}
+
+/** Set a specific layout for a view. No-ops if already in the requested layout. */
+export async function setViewLayout(viewId: ViewId, layout: ViewLayout): Promise<void> {
+  const current = getViewLayout(viewId);
+  if (current === layout) {
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/views/viewLayout.ts]: Fixed in d4f451f — extracted applyViewLayout() so toggle and set each read config once

File: packages/core/package.json [hunk 6]
```diff
@@ -275,6 +355,68 @@
...
       ],
+      "commandPalette": [
+        {
+          "command": "workcenter.switchInboxToTree",
+          "when": "workcenter.inboxLayout == flat"
+        },
+        {
+          "command": "workcenter.switchInboxToFlat",
+          "when": "workcenter.inboxLayout == tree"
+        },
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/package.json:388]: Layout commands only work when a view is visible. The extension already activates on onView:workcenter.* which covers this. Adding onCommand: for 15 commands would bloat activation events without benefit.
